xhydro/modelling/obj_funcs.py

```diff
@@ -0,0 +1,856 @@
+            - "pbias" : Percent bias (relative bias)
+            - "r2" : r-squared, i.e. square of correlation_coeff.
+            - "rmse" : Root Mean Square Error
+            - "rrmse" : Relative Root Mean Square Error (RMSE-to-mean ratio)
+            - "rsr" : Ratio of RMSE to standard deviation.
+            - "volume_error": Total volume error over the period.
+
+        The default is 'rmse'.
+
+    take_negative : bool
+        Used to force the objective function to be multiplied by minus one (-1)
+        such that it is possible to maximize it if the optimizer is a minimizer
+        and vice-versa. Should always be set to False unless required by an
+        optimization setup, which is handled internally and transparently to
+        the user. The default is False.
+
+    mask : array_like
+        Array of 0 or 1 on which the objective function should be applied.
+        Values of 1 indicate that the value is included in the calculation, and
+        values of 0 indicate that the value is excluded and will have no impact
+        on the objective function calculation. This can be useful for specific
+        optimization strategies such as odd/even year calibration, seasonal
+        calibration or calibration based on high/low flows. The default is None
+        and all data are preserved.
+
+    transform : str
+        Indicates the type of transformation required. Can be one of the
+        following values:
+
+            - "sqrt" : Square root transformation of the flows [sqrt(Q)]
+            - "log" : Logarithmic transformation of the flows [log(Q)]
+            - "inv" : Inverse transformation of the flows [1/Q]
+
+        The default value is "None", by which no transformation is performed.
+
+    epsilon : float
+        Indicates the perturbation to add to the flow time series during a
+        transformation to avoid division by zero and logarithmic transformation.
+        The perturbation is equal to:
+
+            perturbation = epsilon * mean(qobs)
+
+        The default value is 0.01.
+
+    Returns
+    -------
+    float
+        Value of the selected objective function (obj_fun).
+    """
+    # List of available objective functions
+    obj_func_dict = {
+        "abs_bias": _abs_bias,
+        "abs_pbias": _abs_pbias,
+        "abs_volume_error": _abs_volume_error,
+        "agreement_index": _agreement_index,
+        "bias": _bias,
+        "correlation_coeff": _correlation_coeff,
+        "kge": _kge,
+        "kge_mod": _kge_mod,
+        "mae": _mae,
+        "mare": _mare,
+        "mse": _mse,
+        "nse": _nse,
+        "pbias": _pbias,
+        "r2": _r2,
+        "rmse": _rmse,
+        "rrmse": _rrmse,
+        "rsr": _rsr,
+        "volume_error": _volume_error,
+    }
+
+    # If we got a dataset, change to np.array
+    if isinstance(qsim, xr.Dataset):
+        qsim = qsim["qsim"]
+
+    if isinstance(qobs, xr.Dataset):
+        qobs = qobs["qobs"]
```

Comment:
Je n'aime pas trop hard-coder ceci, surtout que `streamflow`  est le nom qu'on vise pour le reste de `xhydro`, mais on peut laisser ça comme ça dans cette PR et y revenir.
```suggestion
    # If we got a dataset, change to np.array
    # FIXME: Implement a more flexible method.
    if isinstance(qsim, xr.Dataset):
        qsim = qsim["qsim"]

    if isinstance(qobs, xr.Dataset):
        qobs = qobs["qobs"]
```

Reply:
J'avoue que ça serait mieux que ceci soit plus flexible. Je vais faire appel à votre expertise pour voir quelle option est la meilleure! J'ai mis le fixme en attendant.